### PR TITLE
Code cleanup - handle key not found

### PR DESCRIPTION
**Pull request by ocallesp**
The map Properties may not contain the key value.
The current logic checks for existing key is present in ChangedProperties, but it gets the value from Properties causing to throw an exception.

This change only adds logic to check if Properties contains the key before getting the value.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/project-system/pull/6736)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/ProjectFilePathAndDisplayNameEvaluationHandler.cs [hunk 1]
```diff
@@ -41,9 +41,9 @@ public void Handle(IComparable version, IProjectChangeDescription projectChange,
 
             VerifyInitialized();
 
-            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.MSBuildProjectFullPathProperty))
+            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.MSBuildProjectFullPathProperty) &&
+                projectChange.After.Properties.TryGetKey(ConfigurationGeneral.MSBuildProjectFullPathProperty, out string projectFilePath))
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/ProjectFilePathAndDisplayNameEvaluationHandler.cs:45]:
```suggestion
                projectChange.After.Properties.TryGetValue(ConfigurationGeneral.MSBuildProjectFullPathProperty, out string projectFilePath))
```

**Review by jjmew — APPROVED** [review]

**Comment by davkean** [comment]
This is a pretty important property, without it, language service integration will be broken. I'll like to understand the cause a little more. Please hold off while I investigate this. I suspect a broken project file in which case I would like instead to embed the rule so that this never happens.

**Comment by ocallesp** [comment]
> This is a pretty important property, without it, language service integration will be broken. I'll like to understand the cause a little more. Please hold off while I investigate this. I suspect a broken project file in which case I would like instead to embed the rule so that this never happens.

@davkean Do you mean Embedding the xaml in your assembly and exposing it via a MEF export ?
I see `ConfigurationGeneral.xaml` is already in [Microsoft.VisualStudio.ProjectSystem.Managed.csproj](https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/Microsoft.VisualStudio.ProjectSystem.Managed.csproj#L328)

**Comment by davkean** [comment]
This is a fix that fixes the underlying cause; the rule missing. https://github.com/dotnet/project-system/pull/6740

**Comment by davkean** [comment]
@ocallesp Was this deliberately merged in? Can you revert it in lieu of the the change that fixed the underlying change: https://github.com/dotnet/project-system/pull/6740?

**Pull request merged**